### PR TITLE
docs/variants/msi_z690/firmware-update.md: add more detailed procedure

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

File: docs/variants/msi_z690/firmware-update.md [hunk 1]
```diff
@@ -12,6 +12,46 @@ sudo dmidecode -t bios | grep Version
...
+Both `BIOS boot medium lock` and `Enable SMM BIOS write protection` should be
+unchecked. [UEFI Secure Boot](../../dasharo-menu-docs/device-manager.md#secure-boot-configuration)
+must be disabled as well (uncheck `Attempt Secure Boot` if
+`Current Secure Boot State` does not say `Disabled`). To apply changes you will
+need to reboot.
+
+You can use [Dasharo Tools Suite (DTS)](../../dasharo-tools-suite/releases.md)
+or your own Linux distribution for firmware update. You will need a correct
+flashrom application for that purpose. Linux distributions may not yet have the
+support for the newesy chipsets in flashrom installed via package manager.
```
Comment (pietrushnic) [inline docs/variants/msi_z690/firmware-update.md:28]: newsy?

File: docs/variants/msi_z690/firmware-update.md [hunk 2]
```diff
@@ -12,6 +12,46 @@ sudo dmidecode -t bios | grep Version
...
+[Dasharo Security Options](../..//dasharo-menu-docs/dasharo-system-features.md).
+Both `BIOS boot medium lock` and `Enable SMM BIOS write protection` should be
+unchecked. [UEFI Secure Boot](../../dasharo-menu-docs/device-manager.md#secure-boot-configuration)
+must be disabled as well (uncheck `Attempt Secure Boot` if
+`Current Secure Boot State` does not say `Disabled`). To apply changes you will
+need to reboot.
+
+You can use [Dasharo Tools Suite (DTS)](../../dasharo-tools-suite/releases.md)
+or your own Linux distribution for firmware update. You will need a correct
+flashrom application for that purpose. Linux distributions may not yet have the
```
Comment (pietrushnic) [inline docs/variants/msi_z690/firmware-update.md:27]: how to identify if I have correct flashrom version?

File: docs/variants/msi_z690/firmware-update.md [hunk 3]
```diff
@@ -12,6 +12,46 @@ sudo dmidecode -t bios | grep Version
...
+support for the newesy chipsets in flashrom installed via package manager.
+Procedure for building the right flashrom is described in Build flashrom
+section in the [Initial deployment documentation](initial-deployment.md#initial-deployment-manually).
+DTS is already equipped with a correct flashrom build. You will need to
+download a firmware update binary. When you boot DTS enter the shell from the
```
Comment (pietrushnic) [inline docs/variants/msi_z690/firmware-update.md:32]: I would rephrase whole paragraph to make it very clear. Now it say DTS, Linux distro, DTS - no easy to follow. Something as follows (your content can be easily reused):

Please use one of the following environments to perform Dasharo update:
- DTS - ...
- Linux distribution of your choice  - ...

File: docs/variants/msi_z690/firmware-update.md [hunk 4]
```diff
@@ -12,6 +12,46 @@ sudo dmidecode -t bios | grep Version
...
+
```
Comment (pietrushnic) [inline docs/variants/msi_z690/firmware-update.md:33]: not needed empty line?

File: docs/variants/msi_z690/firmware-update.md [hunk 5]
```diff
@@ -12,6 +12,46 @@ sudo dmidecode -t bios | grep Version
...
+Or for DDR5 platform:
+
+```bash
+wget https://3mdeb.com/open-source-firmware/Dasharo/msi_ms7d25/v1.1.1/msi_ms7d25_v1.1.1_ddr5.rom
+```
+
+They will also work in any Linux terminal with wget installed. Then proceed
+with commands described below.
+
+## Migrating SMBIOS unique data
```
Comment (pietrushnic) [inline docs/variants/msi_z690/firmware-update.md:50]:
```suggestion
## Migrating SMBIOS unique data (optional)
```

File: docs/variants/msi_z690/firmware-update.md [hunk 6]
```diff
@@ -12,6 +12,46 @@ sudo dmidecode -t bios | grep Version
...
+
+Optionally before flashing you may migrate your serial number and UUID as
```
Comment (pietrushnic) [inline docs/variants/msi_z690/firmware-update.md:52]:
```suggestion
Before flashing you may migrate your serial number and UUID as
```

**Review by pietrushnic — CHANGES_REQUESTED** [review]

File: docs/variants/msi_z690/firmware-update.md (hunk 1 above)
Comment (miczyg1) [reply docs/variants/msi_z690/firmware-update.md:28]:
```suggestion
support for the newest chipsets in flashrom installed via package manager.
```

File: docs/variants/msi_z690/firmware-update.md (hunk 2 above)
Comment (miczyg1) [reply docs/variants/msi_z690/firmware-update.md:27]: If one follows our guide, it gets the right flashrom. I ask to build it from our sources in subsequent sentences

File: docs/variants/msi_z690/firmware-update.md (hunk 4 above)
Comment (miczyg1) [reply docs/variants/msi_z690/firmware-update.md]: Yes, removed.

File: docs/variants/msi_z690/firmware-update.md (hunk 2 above)
Comment (miczyg1) [reply docs/variants/msi_z690/firmware-update.md]: Refactored the sections a lot. Please review again

**Comment by miczyg1** [comment]
@pietrushnic also added info about automatic update using an option from DTS menu but waiting for documentation update to link to it

**Comment by miczyg1** [comment]
> @pietrushnic also added info about automatic update using an option from DTS menu but waiting for documentation update to link to it

Documentation update is merged so added the link to it.

File: docs/variants/msi_z690/firmware-update.md [hunk 7]
```diff
@@ -37,3 +162,56 @@ In this case, the whole `bios` region must be updated.
 ```bash
 flashrom -p internal -w [path] --ifd -i bios
 ```
+
+## Troubleshooting
```
Comment (pietrushnic) [inline docs/variants/msi_z690/firmware-update.md:166]: This section is very generic, shouldn't it go to: https://docs.dasharo.com/osf-trivia-list/deployment/#flashrom ?

**Review by pietrushnic — CHANGES_REQUESTED** [review]

File: docs/variants/msi_z690/firmware-update.md (hunk 7 above)
Comment (miczyg1) [reply docs/variants/msi_z690/firmware-update.md:166]: Right... I have moved it to the suggested section

**Review by pietrushnic — APPROVED** [review]
